nextjs-starter/app/examples/get-recipe/Shared.tsx

```diff
@@ -1,88 +1,78 @@
-"use client"
+'use client';
 
-import { getRecipe } from "@/app/actions/streamable_objects"
-import { Button } from "@/components/ui/button"
-import { Input } from "@/components/ui/input"
-import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card"
-import { motion, AnimatePresence } from "framer-motion"
-import { Loader2, Cog, AlertCircle, CheckCircle } from "lucide-react"
-import { StreamState } from "@/app/_hooks/useStream"
-import preloadedExamples from "./examples"
-import { RecipeRender } from "./Recipe"
-import { useState } from "react"
-import { Tabs, TabsContent, TabsList, TabsTrigger } from "@/components/ui/tabs"
-import { ScrollArea } from "@/components/ui/scroll-area"
-import { Badge } from "@/components/ui/badge"
-import JsonView from 'react18-json-view'
+import type { HookResult } from '@/baml_client/react/types';
+import { Button } from '@/components/ui/button';
+import { Card, CardContent, CardHeader, CardTitle } from '@/components/ui/card';
+import { Input } from '@/components/ui/input';
+import { ScrollArea } from '@/components/ui/scroll-area';
+import { AnimatePresence, motion } from 'framer-motion';
+import { AlertCircle, CheckCircle, Cog, Loader2 } from 'lucide-react';
+import { useState } from 'react';
+import JsonView from 'react18-json-view';
+import ErrorPreview from '../_components/ErrorPreview';
+import { RecipeRender } from './Recipe';
+import preloadedExamples from './examples';
 
 export const Content: React.FC<{
-  query: string
-  setQuery: (value: string) => void
-  answer: StreamState<typeof getRecipe>
+  query: string;
+  setQuery: (value: string) => void;
+  answer: HookResult<'GetRecipe'>;
 }> = ({ query, setQuery, answer }) => {
-  const [name, setName] = useState<string>("")
+  const [name, setName] = useState<string>('');
 
   const handleSubmit = async (text: string) => {
-    answer.mutate(text)
-    setName(text)
-  }
+    answer.mutate(text);
+    setName(text);
+  };
 
-  const data = answer.isSuccess
-    ? answer.data
-    : answer.isLoading
-    ? answer.streamingData
-    : undefined
+  const data = answer.isSuccess ? answer.data : answer.isPending ? answer.partialData : undefined;
```

Comment:
i wish we had a helper for this. its hard to do right.

Reply:
Great callout. Maybe we can ship something after we get this v1 out.

Maybe it's another field on the hook result? like partialOrFinal? or something like that